README.md

```diff
@@ -1,120 +1,87 @@
+		Assets,
+		Matcher,
+		AccountIdConverter,
+		AccountId,
+		CheckAsset,
+		CheckingAccount,
+	>
 ```
 
-After this step, you should have a validator node online with a session key for your node.
-For key management and validator rewards, consult our [validator guide online](https://docs.astar.network/build/validator-guide/configure-node).
+In order to make pallet contracts compatible with our new adapter, we have to implement trait `nonfungibles::MutatePSP34<AccountId>` for pallet_contracts.
+Here is the PR for new trait definitions for pallet_contracts: https://github.com/gitofdeepanshu/substrate/pull/1
 
-## Run RPC Tests
+and here is the actual implentation of traits : https://github.com/paritytech/substrate/commit/ab6646e57c8a340d9365833e198f766fefc80ede
 
-RPC tests suite can be run for any release. To run tests go to https://github.com/AstarNetwork/Astar/actions/workflows/rpcTest.yml. Click Run workflow, in the dropdown input the release version tag you want to run the test suite. Then click the green Run workflow button to start the test suite. 
+### MultiAsset Scheme for PSP34 
+To reference a NFT, we need :
 
-![Screenshot from 2022-07-07 15-28-46](https://user-images.githubusercontent.com/874046/177785570-330c6613-237d-4190-bfed-69876209daf6.png)
+a. Smart Contract Address
 
-## Workspace Dependency Handling
+b. Item Id for a specific NFTin collection
 
-All dependencies should be listed inside the workspace's root `Cargo.toml` file.
-This allows us to easily change version of a crate used by the entire repo by modifying the version in a single place.
+```
+MultiAsset {
+id: Concrete( MultiLocation { parents: 0, interior: Junctions::X1(AccountId32 {network: None, id: [u8;32] } ) } ),
+fun: Fungibility::NonFungible::Index(u128)
+```
 
-Right now, if **non_std** is required, `default-features = false` must be set in the root `Cargo.toml` file (related to this [issue](https://github.com/rust-lang/cargo/pull/11409)). Otherwise, it will have no effect, causing your compilation to fail.
-Also `package` imports aren't properly propagated from root to sub-crates, so defining those should be avoided.
+We get smart contract address from `id` and item id from `fun`
 
-Defining _features_ in the root `Cargo.toml` is additive with the features defined in concrete crate's `Cargo.toml`.
+## Analysis
 
-**Adding Dependency**
-1. Check if the dependency is already defined in the root `Cargo.toml`
-    1. if **yes**, nothing to do, just take note of the enabled features
-    2. if **no**, add it (make sure to use `default-features = false` if dependency is used in _no_std_ context)
-2. Add `new_dependecy = { workspace = true }` to the required crate
-3. In case dependency is defined with `default-features = false` but you need it in _std_ context, add `features = ["std"]` to the required crate.
- 
+### What are the potential problems with this approach?
 
-## Further Reading
+Problems can be:
+1. No Gas Limit set for XCM call execution.
```

Comment:
Actually, it is limited :slightly_smiling_face: 

You have to pay for XCM execution up-front, and this will directly translate to the _gas limit_.